indico/modules/events/surveys/__init__.py

```diff
@@ -62,8 +63,10 @@ def _get_active_surveys(event):
     if not event.has_feature('surveys'):
         return []
     from indico.modules.events.surveys.models.surveys import Survey
+    private_criterion = ~Survey.private | Survey.submissions.any(
```

Comment:
Use `db.or_()` when you need to wrap it. It's quite ugly with the `|` operator. Or simply split it:


```
user_pending_criterion= ~SurveySubmission.is_submitted & (SurveySubmission.user == session.user)
private_criterion = ~Survey.private | Survey.submissions.any(user_pending_criterion)
```

actually, if there's no user you can skip the check:

```
private_criterion = ~Survey.private
if session.user:
    user_pending_criterion = ~SurveySubmission.is_submitted & (SurveySubmission.user == session.user)
    private_criterion |= Survey.submissions.any(user_pending_criterion)
```

not sure if `|=` works or you need `private_criterion = private_criterion | ...` (depends on how well sqlalchemy overloads this ;))